tests/unittests/GraphOptzTest.cpp

```diff
@@ -495,35 +495,75 @@ TEST_F(GraphOptz, batchNormAfterConvNotOptimizeWhenMoreThanOneUseOfConv) {
   EXPECT_EQ(conv->getInput().getNode(), A);
 }
 
-TEST_F(GraphOptz, sinkTransposeBelowOptimizeBatchNorm) {
+enum class TestSinkTransposeNodesKind {
+  Batch,
+  Relu,
+  Sigmoid,
+  Tanh,
+  Quantize,
+};
+
+class GraphOptzSinkTransposeBelowParametrized
+    : public GraphOptz,
+      public ::testing::WithParamInterface<TestSinkTransposeNodesKind> {
+public:
+  Node *getNodeFromInput(TestSinkTransposeNodesKind testNode, Node *T) {
+    switch (testNode) {
+    case TestSinkTransposeNodesKind::Batch: {
+      return F_->createBatchNormalization(bindings_, "batch", T, 3, 0.0001,
+                                          0.9);
+    }
+    case TestSinkTransposeNodesKind::Relu: {
+      return F_->createRELU("relu", T);
+    }
+    case TestSinkTransposeNodesKind::Sigmoid: {
+      return F_->createSigmoid("sigmoid", T);
+    }
+    case TestSinkTransposeNodesKind::Tanh: {
+      return F_->createTanh("tanh", T);
+    }
+    case TestSinkTransposeNodesKind::Quantize: {
+      return F_->createQuantize(
+          "quantize", T,
+          mod_.uniqueType(ElemKind::Int8QTy, T->dims(0), 0.03, 5));
+    }
+    default: { return nullptr; }
```

Comment:
nit: since TestSinkTransposeNodesKind is enum class we can drop default here. If one of the values missing in switch, compiler will issue a warning which we treat as error.